lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -1070,6 +1127,122 @@ def plot(self, *, show_outline=True, show_labels=True, axes=None, **kwargs):
+    def plot_fit(self, frame_idx, *, fit_kwargs=None, data_kwargs=None, show_track_index=True):
+        """Plot the localization model fits for a particular kymograph frame.
+
+        Note that currently, this fit is only available for Gaussian refinement.
+
+        Parameters
+        ----------
+        frame_idx : int
+            Index of the kymograph frame to plot.
+        fit_kwargs : dict
+            Dictionary containing arguments to be passed to the fit plot.
+        data_kwargs : dict
+            Dictionary containing arguments to be passed to the data plot.
+        show_track_index : bool
+            Display track index in the plot.
+
+        Raises
+        ------
+        ValueError
+            If the group is empty.
+        NotImplementedError
+            If this group has multiple kymographs associated with it.
+        NotImplementedError
+            If localization was not performed using Gaussian localization.
+        IndexError
+            If requesting a fit for a frame index out of bounds of the KymoTrack.
+
+        Examples
+        --------
+        ::
+
+            import lumicks.pylake as lk
+
+            tracks = lk.track_greedy(kymo, channel="red", pixel_threshold=5)
+            refined = lk.refine_tracks_gaussian(tracks, window=10, refine_missing_frames=True, overlap_strategy="multiple")
+
+            # Plot the localization fits for kymograph frame 5
+            refined.plot_fit(5)
+
+            # Using ipywidgets and jupyter, it is easy to make an interactive version of this plot.
+            import ipywidgets as widgets
+
+            plt.figure()
+            def plot_fit(frame):
+                plt.cla()  # Clear the current axis
+                refined.plot_fit(frame, show_track_index=True)
+
+            widgets.interact(plot_fit, frame=widgets.IntSlider(0, max=kymo.shape[1] - 1));
+        """
+        self._validate_single_source("plot_fit")
+        if not self._kymos:
+            raise ValueError("No tracks in the group")
+
+        kymo = self._kymos[0]  # We can rely on there being one (_validate_single_source)
+        num_frames = kymo.shape[1]
+        frame_idx = num_frames + frame_idx if frame_idx < 0 else frame_idx
+        if not 0 <= frame_idx < num_frames:
+            raise IndexError(
+                f"Frame index is out of range of the kymograph. Kymograph length is {num_frames}."
+            )
+
+        kymo_data = kymo.get_image(self._channel)[:, frame_idx]
+        data_kwargs = {"color": "#c8c8c8", **(data_kwargs or {})}
+        plt.plot(np.arange(kymo.pixels_per_line) * kymo.pixelsize[0], kymo_data, **data_kwargs)
+
+        # Put the numeric values above the baseline
+        y_coord = np.min(kymo_data) + 0.05 * (np.max(kymo_data) - np.min(kymo_data))
+        for track_idx, track_frame_idx in self._tracks_in_frame(frame_idx):
+            self[track_idx].plot_fit(
+                track_frame_idx, show_data=False, show_legend=False, fit_kwargs=fit_kwargs
```

Comment:
why do we need `show_legend=False`?

Reply:
Premature optimization. Didn't want to constantly set the `xlabels`, but I just tested a bit and it doesn't matter much at all. I dropped the `show_legend` arg now.